enhancements/sig-policy/72-content-to-ansilbe-job/README.md

```diff
@@ -0,0 +1,161 @@
+
+Current `extra_vars` has a field `target_clusters` contains non-compliant cluster name list:
+```json
+{
+  "target_clusters": [
+    "cluster1", 
+    "cluster2", 
+    "cluster3",
+  ]
+}
+```
+
+The new content structure will pass more information:
+```json
+{
+  "target_clusters": [
+    "cluster1", 
+    "cluster2", 
+    "cluster3",
+  ],
+  "policy_name": "policy1",
+  "violation_message": "NonCompliant; violation -  xxxxxx",
+  "policy_violation_context" : {
+    "cluster1": "replicated policy status1",
+    "cluster2": "replicated policy status2",
+    },
+}
+```
+
+- `policy_name` is the name of the non-compliance policy that triggers the ansible job.
+  `violation_message` is the latest policy violation message. 
+  They are from the root policy on the hub cluster. 
+  The field path of `policy_name` is `policy.metadata.name`.
+  The field path of `violation_message` is `policy.policyStatus.details[0].history[0].message`
+
+- `policy_violation_context` is a map. The keys are non-compliant cluster names, and the value is the trimmed replicated 
+  policy status for each non-compliant cluster.
+  The policy automation looks at the `cluster namespaces` of the hub cluster and gets the replicated policies. Then it filters out the non-compliance replicated policies and puts the trimmed status into the `policy_violation_context` map.
+
+  Suppose the policy automation runs on the managed cluster. In that case, the policy automation looks at the propagated policy on the managed cluster, then `policy_violation_context` would be a map that only contains the propagated policy status.
+
+  The field path of the raw status is `policy.policyStatus`. Then the trimmed status will skip the below unnecessary fields:
+  1. `policy.policyStatus.details[n].templateMeta`.
+  2. `policy.policyStatus.details[n].history[n].eventName`. Here we can consider only keep the latest policy violation history, 
+  which are `policy.policyStatus.details[0].history[0].lastTimestamp` and `policy.policyStatus.details[0].history[0].message`
+
+### User Stories
+
+#### Story 1
+
+As an OCM Policy addon user, I would like to get the additional status in an Ansible job's `extra_vars` field. For example,
+regarding the CertificatePolicy policy, which checks for expiring certificates in secrets, I want to get the replicated policy 
+status for each non-compliant cluster so I can know which secret expires on a specific cluster in the Anisble Job. Then, the 
+Ansible Automation Platform can create a notice ticket and provide the violation details. 
+
+### Implementation Details/Notes/Constraints [optional]
+
+N/A
+
+### Risks and Mitigation
+
+Since `policy_violation_context` is the replicated policy status for all non-compliant managed clusters, the length of 
+`policy_violation_context` may be oversized when the amount of managed clusters is large.
+
+A possible solution only passes the top `N` replicated policy status into `policy_violation_context`, `N` is configurable
```

Comment:
I think we probably want to specify how we are going to select the "top" N policy statuses to pass into the context. I'm not sure how we would prioritize which replicated policies make it in, so I think if there are more than "N" policies we would probably want to not include the replicated policy statuses at all.

Reply:
Also, would we configure this with an env variable on the propagator?

Reply:
> I think we probably want to specify how we are going to select the "top" N policy statuses to pass into the context. I'm not sure how we would prioritize which replicated policies make it in, so I think if there are more than "N" policies we would probably want to not include the replicated policy statuses at all.

It is another solution,  we can discuss it if we decide such configure is needed.

Reply:
> Also, would we configure this with an env variable on the propagator?

I prefer it to be a field of policy automation.